spatial_maps/maps.py

```diff
@@ -78,11 +81,19 @@ def occupancy_map(self, smoothing):
         return smooth_map(self.time_pos, self.bin_size, smoothing)
 
     def rate_map(self, smoothing, mask_zero_occupancy=False):
-        spike_map = self.spike_map(smoothing)
-        occupancy_map = self.occupancy_map(smoothing)
+        spike_map = self.spike_map(smoothing=0)
+        occupancy_map = self.occupancy_map(smoothing=0)
         if mask_zero_occupancy:
             # to avoid infinity (x/0) we set zero occupancy to nan
             # this can be handy when analyzing low occupancy maps
-            occupancy_map[self.time_pos == 0] = np.nan
-        rate_map = spike_map / occupancy_map
+            ocmap = copy(occupancy_map)
+            ocmap[self.time_pos == 0] = np.nan
+            rate_map = spike_map / ocmap
+            rate_map = smooth_map(rate_map, self.bin_size, smoothing)
+            rate_map[self.time_pos == 0] = np.nan
+        else:
+            rate_map = spike_map / occupancy_map
+            rate_map[np.isinf(rate_map)] = 0
+            rate_map[np.isnan(rate_map)] = 0
+            rate_map = smooth_map(rate_map, self.bin_size, smoothing)
```

Comment:
Also here you might get inf when dividing by the oc map. When mask_zero_occupancy is false then I set them to 0.
Finally I smooth the map

Reply:
This is not a problem when smoothing both maps before dividing, I think.

Reply:
I think it is as you might still get Inf of you have a large area of unoccupied spaces like the square track

Reply:
True, but is not an inf then better than a zero, or at least nan, as an indication to the user that it is not possible to calculate the rate there. Btw, you may use `np.isfinite` which catches both nan and inf

Reply:
decided: I leave it as it is and mask it outside :)

Reply:
:+1: